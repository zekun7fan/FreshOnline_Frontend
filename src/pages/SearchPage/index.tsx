import React, {useState} from "react";
import {Layout, Breadcrumb} from 'antd';

import SearchHeaderUI from "../../components/SearchHeader";
import SearchSiderUI from "../../components/SearchSider";
import SearchContentUI from "../../components/SearchContent";
import {Link} from "react-router-dom";

const { Header, Content, Sider, Footer } = Layout;

export default function SearchPage() {
    return (
        <Layout>
            <Layout>
                <Sider width={200} className="site-layout-background">
                    <SearchSiderUI />
                </Sider>
                <Layout style={{ padding: '0 24px 24px' }}>
                    <Breadcrumb style={{ margin: '16px 0' }}>
                        <Breadcrumb.Item><Link to='/'>Home</Link></Breadcrumb.Item>
                        <Breadcrumb.Item>Search</Breadcrumb.Item>
                    </Breadcrumb>
                    <Content className="site-layout-background"
                        style={{
                            padding: 24,
                            margin: 0,
                            minHeight: 280,
                        }}>
                        <SearchContentUI/>
                    </Content>
                </Layout>
            </Layout>
            <SearchFooter/>
        </Layout>
    );
}

export function SearchFooter() {
    return (
        <Layout>
            <Footer style={{ textAlign: 'center' }}>FreshOnline Â©2022 Created by Team FreshOnline</Footer>
        </Layout>
    )
}