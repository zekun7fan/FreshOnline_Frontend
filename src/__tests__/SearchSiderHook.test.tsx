import {renderHook, act, cleanup} from '@testing-library/react-hooks';

import SearchSiderUI, {SearchSider} from "../components/SearchSider";
import SearchContentUI from "../components/SearchContent";
import SearchPage from "../pages/SearchPage";
import SearchHeaderUI from "../components/SearchHeader";
import store from "../redux/store";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";

jest.mock('antd/es/tree-select', () => ({}));

afterEach(() => {
    cleanup().then()
})

test('render search sider using hook', () => {
    const wrapper = ({ children }:{ children: any }) => (
        <Provider store={store}>
            <BrowserRouter>
                {children}
            </BrowserRouter>
        </Provider>
    )
    const { result } = renderHook(() => SearchSider(), { wrapper })
    // console.log(result.current)
})

test('render search sider UI using hook', () => {
    const wrapper = ({ children }:{ children: any }) => (
        <Provider store={store}>
            <BrowserRouter>
                {children}
            </BrowserRouter>
        </Provider>
    )
    const { result } = renderHook(() => SearchSiderUI(), { wrapper })
    expect(result.current.props.children.length).toEqual(3)
    expect(typeof result.current.props.children[0].type).toBe('function')
    expect(typeof result.current.props.children[1].type).toBe('function')
    expect(typeof result.current.props.children[2].type).toBe('object')
    expect(result.current).toMatchSnapshot()
})