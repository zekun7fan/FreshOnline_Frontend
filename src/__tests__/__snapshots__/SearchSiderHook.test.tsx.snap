// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render search sider UI using hook 1`] = `
<Content
  className="sider-content"
  style={
    Object {
      "background": "white",
      "minHeight": 1710,
    }
  }
>
  <Menu
    defaultOpenKeys={
      Array [
        "sub1",
        "sub2",
        "sub3",
      ]
    }
    mode="inline"
    style={
      Object {
        "borderRight": 0,
        "height": "100%",
      }
    }
  >
    <SubMenu
      icon={<ForwardRef(UserOutlined) />}
      title="Brands"
    >
      <Memo
        onChange={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
        value={Array []}
      >
        <Row />
      </Memo>
    </SubMenu>
    <SubMenu
      icon={<ForwardRef(UserOutlined) />}
      title="Price Selection"
    >
      <Row>
        <Col
          offset={4}
        >
          <ForwardRef
            addonBefore="Min"
            max={9999}
            min={0}
            onChange={[Function]}
            precision={0}
            value={0}
          />
        </Col>
        <Col
          offset={4}
        >
          <ForwardRef
            addonBefore="Max"
            max={9999}
            min={0}
            onChange={[Function]}
            precision={0}
            value={0}
          />
        </Col>
      </Row>
    </SubMenu>
    <SubMenu
      icon={<ForwardRef(UserOutlined) />}
      title="Sort Type"
    >
      <Memo
        onChange={[Function]}
        value={0}
      >
        <Row
          gutter={
            Array [
              0,
              8,
            ]
          }
        >
          <Col
            offset={4}
            span={20}
          >
            <Radio
              value={1}
            >
              Price: Low to High
            </Radio>
          </Col>
          <Col
            offset={4}
            span={20}
          >
            <Radio
              value={2}
            >
              Price: High to Low
            </Radio>
          </Col>
          <Col
            offset={4}
            span={20}
          >
            <Radio
              value={4}
            >
              Top Sellers
            </Radio>
          </Col>
        </Row>
      </Memo>
    </SubMenu>
  </Menu>
  <Divider
    dashed={true}
  />
  <Row
    justify="center"
  >
    <Button
      onClick={[Function]}
      type="primary"
    >
      Apply Filter
    </Button>
  </Row>
</Content>
`;
